### PR TITLE
added encodeURIComponent function interface and tests

### PR DESCRIPTION
**Pull request by Bohemus307**
### Added tests for function in run-tests

### CODE REVIEW

File: devs/run-tests/10encodeURIComponent.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import { assert } from "@devicescript/core"
+import {
+    encodeURIComponent,
+} from "@devicescript/core/src/encodeURIComponent"
```
Comment (Bohemus307) [inline devs/run-tests/10encodeURIComponent.ts:4]: Is this the same situation as the Map class where it needs to be in runtime? Or is this the correct approach for a export function?

File: packages/core/src/encodeURIComponent.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        return (
+            (code >= 48 && code <= 57) || // 0-9
+            (code >= 65 && code <= 90) || // A-Z
+            (code >= 97 && code <= 122) || // a-z
+            code === 45 || // -
+            code === 46 || // .
+            code === 95 || // _
+            code === 126 // ~
+        )
+    }
```
[the remark this answers is not in the document]
Comment (pelikhan) [reply packages/core/src/encodeURIComponent.ts:15]: sounds good; we don't have regexes yet

File: devs/run-tests/10encodeURIComponent.ts (hunk 1 above)
Comment (pelikhan) [reply devs/run-tests/10encodeURIComponent.ts:4]: yes it is the same situtation. I'll chat with Michal about this. We might want to put them in runtime in the meanwhile until get things figured out. You're faster than us!

Comment (Bohemus307) [reply devs/run-tests/10encodeURIComponent.ts:4]: Enjoying the challenges great way to break up the work day.

Comment (Bohemus307) [reply devs/run-tests/10encodeURIComponent.ts]: Ok, I moved the function to runtime and moved tests as well, but I see the submodule issues with jacdac again. If there's anything I'm doing to cause this feel free to let me know hate having such bumpy PR's!

Comment (pelikhan) [reply devs/run-tests/10encodeURIComponent.ts]: we'll fix the submodule (and we should write a guide bout those)

**Comment by pelikhan** [comment]
@Bohemus307 when you're ready, i will take care of the merge/fix with the submodules.

File: devs/run-tests/10encodeURIComponent.ts (hunk 1 above)
Comment (Bohemus307) [reply devs/run-tests/10encodeURIComponent.ts]: Let me know if i can help.

**Comment by Bohemus307** [comment]
Ok @pelikhan, it should be good now but please let me know.

**Review by pelikhan — APPROVED** [review]

**Comment by pelikhan** [comment]
It's in! Thank you!

**Pull request merged**